runtime/fastly/host-api/host_api_fastly.h

```diff
@@ -360,6 +366,74 @@ struct BackendConfig {
   std::optional<HostString> sni_hostname;
   std::optional<ClientCert> client_cert;
   std::optional<bool> grpc;
+  std::optional<uint32_t> http_keepalive_time_ms;
+  std::optional<TcpKeepalive> tcp_keepalive;
+
+  BackendConfig clone() {
```

Comment:
It's too bad this code has to be written manually -- it looks like the struct could *almost* use the default copy constructor except that `HostString` deletes its copy-constructor. I don't know enough about modern C++ idioms to say why but this might be something to see about?

Alternately, perhaps we could write an initializer-list expression (`return BackendConfig { a, b, c, ... }`) and use a helper to clone `std::optional<HostString>` so that at least we'll get a compile error if we add a field and forget to update this?

Reply:
I believe we disabled the copy constructor to ensure a consistent ownership model for host strings. Good suggestion wrt the initializer, I've gone ahead and added that.